### PR TITLE
Implemented basic items system

### PR DESCRIPTION
**Pull request by intsrc**
# REVIEWERS

- [ ] @sonishc 
- [ ] @vlodkopits 

#

# DESCRIPTION

Implemented basic items system with fight integration. Refactored code 

#

# EXPERT REVIEW

- [ ] @rokor

### CODE REVIEW

File: app/assets/javascripts/components/fight.jsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Fight extends React.Component {
-  
-  constructor() {
-    super();
-    
+
+  constructor(props) {
+    super(props);
     this.state = {
+      user: this.setStats(JSON.parse(props.user)),
```
Comment (rokor) [inline app/assets/javascripts/components/fight.jsx]: should be inside const as well as lines 36-37

File: app/assets/javascripts/components/fight.jsx [hunk 2]
```diff
@@ -120,52 +143,108 @@ class Fight extends React.Component {
...
-    if (this.props.bot.hp <=0 || this.props.user.hp <=0){
-      // Send to database EXP
-    }else {
+    const user = this.state.user;
+    const bot = this.props.bot;
+    let EXP = 0;
+    let winner = 'BOT';
+
+    if (bot.hp <=0 || user.hp <=0) {
+      switch (true) {
```
Comment (rokor) [inline app/assets/javascripts/components/fight.jsx]: should be if-else, not switch-case

File: app/models/inventory.rb [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+  belongs_to :user
+  belongs_to :item
   validates_uniqueness_of :item_id, scope: :user_id
   validate :maximum_items_not_more_then_15, on: :create
 
   private
 
   def maximum_items_not_more_then_15
-    errors.add(:base, 'Your backpack is full!') if user_id.count > 15
+    errors.add(:base, 'Your backpack is full!') if user.items.count > 15
```
Comment (omaik) [inline app/models/inventory.rb]: 1. Please forget about the magic numbers. Use constant instead. And rename method to something more simple and understandable. f.e. `check_items_limit` or just `items_limit`.
2. Think about using `counter_cache` if you often need items count.

File: app/models/user.rb [hunk 4]
```diff
@@ -22,4 +24,20 @@ def handle_bot_hp(user)
   def convert_email_to_downcase
     self.email = email.downcase
   end
+
+  def set_items
+    Item.all.each do |item|
```
Comment (omaik) [inline app/models/user.rb]: I am not aware of your business logic, but guess these lines are not correct.

File: app/models/user.rb [hunk 5]
```diff
@@ -3,15 +3,17 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
-  belongs_to :role
+  #belongs_to :role
```
Comment (rokor) [inline app/models/user.rb:6]: if you need to remove this do that not just commented it out

File: app/models/user.rb [hunk 6]
```diff
@@ -22,4 +24,20 @@ def handle_bot_hp(user)
...
   end
+
+  def set_items
+    Item.all.each do |item|
+      items << item
+    end
+  end
+
+  def calculate_stats
+    self.stamina = 0
```
Comment (rokor) [inline app/models/user.rb:35]: you do not really need to use self all over the place

**Pull request closed without merging**